source/credentials_provider_ecs.c

```diff
@@ -629,3 +628,93 @@ struct aws_credentials_provider *aws_credentials_provider_new_ecs(
 
     return NULL;
 }
+
+struct aws_credentials_provider *aws_credentials_provider_new_ecs_from_environment(
+    struct aws_allocator *allocator,
+    const struct aws_credentials_provider_ecs_environment_options *options) {
+
```

Comment:
aws_raise_error(AWS_ERROR_INVALID_ARGUMENT) if anything is missing from options-struct. tls_ctx is the specifically the one I'd be worried about someone accidentally omitting.

Should we **require** tls_ctx? If it's optional, check later after reading the FULL_URI, in case it's "https://". Maybe simpler to just make it required. So someone doesn't build a program that **seems** to work until it's in an environment where FULL_URI is "https://"

Reply:
Thanks, I have made it a required parameter. We always have the tls_ctx whenever we need this API; so it's simpler to just always require it.